dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java

```diff
@@ -1538,4 +1650,342 @@ public void removeMemberUnprocessableTest() throws Exception {
             }
         }
     }
+
+    @Test
+    public void deleteGroupTest() throws Exception {
+
+        GroupService groupService = EPersonServiceFactory.getInstance().getGroupService();
+
+        Group parentGroup = null;
+
+        try {
+            context.turnOffAuthorisationSystem();
+
+            parentGroup = groupService.create(context);
```

Comment:
You should be using the `GroupBuilder.createGroup()` method to create *all the Groups* in these tests.  That way they will be automatically cleaned up after the test completes. In other words, you no longer need *any* of the `finally` clauses in any of the tests you created, as long as you use the `GroupBuilder` test utility.